File: unittest/scripts/js_dev_api_examples/results/Working_with_SQL_Result_Sets_2.js

function print_result(res) {
  if (res.hasData()) {
    // SELECT
    var columns = res.getColumns();
    var record = res.fetchOne();
    
    while (record){
      for (index in columns){
        print (columns[index].getColumnName() + ": " + record[index] + "\n");
      }
      
      // Gets the next record
      record = res.fetchOne();
    }
    
  } else {
    // INSERT, UPDATE, DELETE, ...
    print('Rows affected: ' + res.getAffectedRowCount());
  }
}

print_result(nodeSession.sql('DELETE FROM users WHERE age > 40').execute());
print_result(nodeSession.sql('SELECT * FROM users WHERE age = 40').execute());

File: unittest/scripts/py_dev_api_examples/results/Working_with_SQL_Result_Sets_2.py
def print_result(res):
  if res.has_data():
    # SELECT
    columns = res.get_columns()
    record = res.fetch_one()
    
    while record:
      index = 0
      
      for column in columns:
        print "%s: %s \n" % (column.get_column_name(), record[index])
        index = index + 1
      
      # Get the next record
      record = res.fetch_one()
  else:
    #INSERT, UPDATE, DELETE, ...
    print 'Rows affected: %s' % res.get_affected_items_count()

print_result(mySession.sql('DELETE FROM users WHERE age > 40').execute())
print_result(mySession.sql('SELECT * FROM users WHERE age = 40').execute())
